src/python/pants/init/plugin_resolver.py

```diff
@@ -1,201 +1,140 @@
-import shutil
 import site
-import uuid
-from typing import Iterable, Iterator, List, Optional, Type, TypeVar, cast
+import sys
+from typing import Optional, TypeVar, cast
 
-from pex import resolver
-from pex.interpreter import PythonInterpreter
-from pex.network_configuration import NetworkConfiguration
-from pkg_resources import Distribution, WorkingSet
+from pkg_resources import WorkingSet
 from pkg_resources import working_set as global_working_set
 
+from pants.backend.python.util_rules.pex import (
+    PexInterpreterConstraints,
+    PexRequest,
+    PexRequirements,
+    VenvPex,
+    VenvPexProcess,
+)
+from pants.core.util_rules.pants_environment import PantsEnvironment
+from pants.engine.collection import DeduplicatedCollection
+from pants.engine.internals.session import SessionValues
+from pants.engine.process import ProcessCacheScope, ProcessResult
+from pants.engine.rules import Get, QueryRule, collect_rules, rule
+from pants.init.bootstrap_scheduler import BootstrapScheduler
 from pants.option.global_options import GlobalOptions
-from pants.option.optionable import Optionable
 from pants.option.options_bootstrapper import OptionsBootstrapper
-from pants.option.scope import ScopeInfo
 from pants.option.subsystem import Subsystem
-from pants.python.python_repos import PythonRepos
-from pants.util.contextutil import temporary_dir
-from pants.util.dirutil import safe_concurrent_rename, safe_delete, safe_open
-from pants.util.memo import memoized_property
-from pants.version import PANTS_SEMVER
+from pants.util.logging import LogLevel
 
 logger = logging.getLogger(__name__)
 
 
 S = TypeVar("S", bound=Subsystem)
 
 
-class PluginResolver:
-    @staticmethod
-    def _is_wheel(path) -> bool:
-        return os.path.isfile(path) and path.endswith(".whl")
+class ResolvedPluginDistributions(DeduplicatedCollection[str]):
+    sort_input = True
+
+
+@rule
+async def resolve_plugins(
+    interpreter_constraints: PexInterpreterConstraints, global_options: GlobalOptions
+) -> ResolvedPluginDistributions:
+    """This rule resolves plugins using a VenvPex, and exposes the absolute paths of their dists.
+
+    NB: This relies on the fact that PEX constructs venvs in a stable location (within the
+    `named_caches` directory), but consequently needs to disable the process cache: see the
+    ProcessCacheScope reference in the body.
+    """
+    requirements = PexRequirements(sorted(global_options.options.plugins))
+    plugins_pex = await Get(
+        VenvPex,
+        PexRequest(
+            output_filename="pants_plugins.pex",
+            internal_only=True,
+            requirements=requirements,
+            interpreter_constraints=interpreter_constraints,
+            # The repository's constraints are not relevant here, because this resolve is mixed
+            # into the Pants' process' path, and never into user code.
+            apply_requirement_constraints=False,
+            description=f"Resolving plugins: {requirements}",
```

Comment:
The logging for this is weird if you have no plugins. 

More generally, I think this rule can early return to avoid building a Pex that isn't needed. In a followup, could you please add:

```python
if not requirements:
    return ResolvedPluginDistributions([])
```